management/data_bagger/src/astrobee_recorder.cc

```diff
@@ -72,570 +75,166 @@ using std::vector;
-
-          if (!c.isFault() && resp2.valid() && resp2.size() > 0 && static_cast<int>(resp2[0]) == 1) {
-            for (int i = 0; i < resp2[2].size(); i++) {
-              if (shouldSubscribeToTopic(resp2[2][i][0], true))
-                subscribe(resp2[2][i][0]);
-            }
-          } else {
-            ROS_ERROR("Node at: [%s] failed to return subscriptions.", static_cast<std::string>(resp[2]).c_str());
-          }
-        }
-      }
-    }
-}
-
-void Recorder::doTrigger() {
-    ros::Publisher pub = node_handle_.advertise<std_msgs::Empty>("snapshot_trigger", 1, true);
-    pub.publish(std_msgs::Empty());
-
-    ros::Timer terminate_timer = node_handle_.createTimer(ros::Duration(1.0), boost::bind(&ros::shutdown));
-    ros::spin();
-}
-
-bool Recorder::scheduledCheckDisk() {
-    boost::mutex::scoped_lock lock(check_disk_mutex_);
-
-    if (ros::WallTime::now() < check_disk_next_)
-        return true;
-
-    check_disk_next_ += ros::WallDuration().fromSec(20.0);
-    return checkDisk();
-}
-
-bool Recorder::checkDisk() {
-#if BOOST_FILESYSTEM_VERSION < 3
-    struct statvfs fiData;
-    if ((statvfs(bag_.getFileName().c_str(), &fiData)) < 0) {
-        ROS_WARN("Failed to check filesystem stats.");
-        return true;
-    }
-    uint64_t free_space = 0;
-    free_space = (uint64_t) (fiData.f_bsize) * (uint64_t) (fiData.f_bavail);
-    if (free_space < options_.min_space) {
-        ROS_ERROR("Less than %s of space free on disk with %s.  Disabling recording.",
-                   options_.min_space_str.c_str(), bag_.getFileName().c_str());
-        writing_enabled_ = false;
-        return false;
-    } else if (free_space < 5 * options_.min_space) {
-        ROS_WARN("Less than 5 x %s of space free on disk with %s.",
-                  options_.min_space_str.c_str(), bag_.getFileName().c_str());
-    } else {
-        writing_enabled_ = true;
-    }
-#else
-    boost::filesystem::path p(boost::filesystem::system_complete(bag_.getFileName().c_str()));
-    p = p.parent_path();
-    boost::filesystem::space_info info;
-    try {
-        info = boost::filesystem::space(p);
-    }
-    catch (boost::filesystem::filesystem_error &e) {
-        ROS_WARN("Failed to check filesystem stats [%s].", e.what());
-        writing_enabled_ = false;
-        return false;
-    }
-    if (info.available < options_.min_space) {
-        ROS_ERROR("Less than %s of space free on disk with %s.  Disabling recording.",
-                   options_.min_space_str.c_str(), bag_.getFileName().c_str());
-        writing_enabled_ = false;
-        return false;
-    } else if (info.available < 5 * options_.min_space) {
-        ROS_WARN("Less than 5 x %s of space free on disk with %s.",
-                  options_.min_space_str.c_str(), bag_.getFileName().c_str());
-        writing_enabled_ = true;
-    } else {
-        writing_enabled_ = true;
-    }
-#endif
-    return true;
-}
+    execl("/bin/sh", "sh", "-c", command, NULL);
```

Comment:
Can the user passes some weird string as a topic that would get parsed to evaluate an arbitrary command? Would be better if we could do exec on rosbag record directly without letting things get interpreted by the shell.